inc/datahandlers/user.php

```diff
@@ -1084,11 +1084,11 @@ function insert_user()
 			"timeformat" => $db->escape_string($user['timeformat']),
 			"regip" => $db->escape_binary($user['regip']),
 			"language" => $db->escape_string($user['language']),
-			"showcodebuttons" => $user['options']['showcodebuttons'],
-			"sourceeditor" => $user['options']['sourceeditor'],
-			"away" => $user['away']['away'],
-			"awaydate" => $user['away']['date'],
-			"returndate" => $user['away']['returndate'],
+			"showcodebuttons" => (int)$user['options']['showcodebuttons'],
+			"sourceeditor" => (int)$user['options']['sourceeditor'],
+			"away" => (int)$user['away']['away'],
+			"awaydate" => (int)$user['away']['date'],
+			"returndate" => (int)$user['away']['returndate'],
```

Comment:
Not only digits, I don't think `(int)` will keep the `-`:
<img src="http://i.imgur.com/nbI9Z2I.png" alt="" />